### PR TITLE
TravisCI - move CI to xenial: get not-so outdated packages.

### PR DESCRIPTION
**Pull request by fghoussen**
I believe it would be a good thing to move CI to xenial to get "more updated" packages.
autotools jobs break: circular dependency. Don't know how to fix this. If somebody can help...

### CODE REVIEW

**Comment by fghoussen** [comment]
This PR is related to #157

**Comment by sylvestre** [comment]
Could you please see if we could run travis on the current Ubuntu + xenial?
This because arpack is used on old systems and I would like to keep some CI on "old" systems.
Thanks

**Comment by fghoussen** [comment]
> Could you please see if we could run travis on the current Ubuntu + xenial?

That's what this PR is meant to (try to see if it's possible or not)

> This because arpack is used on old systems and I would like to keep some CI on "old" systems.

You mean 2 CIs ? one for ubunty trusty + one for unbuntu xenail if possible (with matrix in yml ?)

**Comment by sylvestre** [comment]
correct

**Comment by fghoussen** [comment]
OK, I can first try to test "stuffs" to see if this makes ubuntu + xenial OK.

If xenial turns out to be OK, then it would be possible to add matrix in yml to get CI on both ubuntu xenial and trusty (or precise ?)

**Comment by fghoussen** [comment]
Still failing ?!...

```

make[2]: Circular ../../debug_c.h <- ../../debug_c.h dependency dropped.
cd . && /bin/bash ./config.status ../../debug_c.h
config.status: error: invalid argument: `../../debug_c.h'
Makefile:468: recipe for target '../../debug_c.h' failed

```

Don't get why ?!...

**Comment by fghoussen** [comment]
Who Wants to Be a Millionaire ?
Q: How to fix these circular dependencies ?
A: Asking for a friend...

**Comment by fghoussen** [comment]
@sylvestre: note that e3ce42e does not seem to be _exactly_ a "make distcheck". My understanding is that it makes a tarball but doesn't re-deploy/test it (like distcheck does as far as I understand)

**Comment by fghoussen** [comment]
Giving up: no solution for now...

**Comment by fghoussen** [comment]
@sylvestre : ignoring the crap, sounds good to you ? want to keep e3ce42e ?

**Comment by fghoussen** [comment]
still need to include matrix in yml for precise. not sure how to do that

**Comment by fghoussen** [comment]
CMakeLists.txt is supported on not-so-outdated xenial, but *not* on outdated trusty... great...

**Comment by fghoussen** [comment]
@sylvestre : you may allow concurrent jobs in travis ci setting at opencollab side, would get faster ci !

**Comment by fghoussen** [comment]
fixed up to 2 commits : distcheck fix + travis modif

File: .travis.yml [hunk 1]
```diff
@@ -23,15 +23,55 @@ addons:
...
+    - stage: precise-autotools-seq
+      dist: precise
+      script: ./bootstrap && ./configure                                                                && make VERBOSE=1 && make check && make distcheck;
+    - stage: precise-autotools-mpi
+      dist: precise
+      script: ./bootstrap && ./configure --enable-mpi                                                   && make VERBOSE=1 && make check && make distcheck;
+    # trusty <=> test "older" systems, without ICB
+    - stage: trusty-cmake-seq
+      dist: trusty
+      script: mkdir -p build && cd build && cmake -D EXAMPLES=ON -D MPI=OFF -D ICB=OFF               .. && make VERBOSE=1 && make test  && make package_source;
```
Comment (fghoussen) [inline .travis.yml]: CMake "make package_source" <=> autotools "make dist" only, **not** "make dist**check**". You may drop that ?

**Comment by fghoussen** [comment]
@sylvestre : note that circular dependencies seem not to be solved yet (still appear in traces). But at least they no more break the CI which could have been extended to precise, trusty and xenial.

**Comment by fghoussen** [comment]
Should be all good. Need to merge this before #157

**Pull request merged**